### ISSUE TITLE
CIDER trace should return inspectable data (not text)

### ISSUE THREAD

**Issue by bsless** [clojure-emacs/cider#3737]
Following the discussion in https://clojurians.slack.com/archives/C0617A8PQ/p1725002128419939

I think the best API for CIDER's trace would be to return data interactively via the inspector, rather than text which gets unwieldy

**Describe the solution you'd like**

Instead of a traced function printing to the REPL window, users will be able to call `cider-inspect-trace`. Select from traced functions? Then the trace as data will open via the inspector.

**Comment by bbatsov** [comment]
Yeah, something along those lines would be good. Alternatively we can have some defcustom to directly open the trace output in the inspector instead of printing it to the REPL or keep a registry of the traced evaluation (e.g. the last 10 or something) and be able to select one of them to inspect it. 

I'm sure @alexander-yakushev will have some interesting ideas on the front, as our Master of Inspection & Tracing.
